prov/rxm/src/rxm_conn.c

```diff
@@ -786,22 +786,109 @@ int rxm_cmap_alloc(struct rxm_ep *rxm_ep, struct rxm_cmap_attr *attr)
 	return ret;
 }
 
+static ssize_t rxm_send_credits(struct fid_ep *ep, size_t credits)
+{
+	struct rxm_conn *rxm_conn =
+		container_of(ep->fid.context, struct rxm_conn, handle);
+	struct rxm_ep *rxm_ep = rxm_conn->handle.cmap->ep;
+	struct rxm_deferred_tx_entry *def_tx_entry;
+	struct rxm_tx_base_buf *tx_buf;
+	struct iovec iov;
+	struct fi_msg msg;
+	ssize_t ret;
+
+	tx_buf = (struct rxm_tx_base_buf *) rxm_tx_buf_alloc(
```

Comment:
Every place that rxm_tx_buf_alloc() is called is preceded by a cast to some other structure.  This tells me that rxm_tx_buf_alloc() should really just return a void *.  This should be a separate change, and I can add that after merging this.